bundles/org.openhab.binding.wizlighting/src/main/java/org/openhab/binding/wizlighting/internal/discovery/WizLightingDiscoveryService.java

```diff
@@ -0,0 +1,168 @@
+     */
+    public WizLightingDiscoveryService() throws IllegalArgumentException {
+        super(SUPPORTED_THING_TYPES, DISCOVERY_TIMEOUT_SECONDS);
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypes() {
+        return SUPPORTED_THING_TYPES;
+    }
+
+    @Override
+    protected void startScan() {
+        DatagramSocket dsocket = null;
+        try {
+            RegistrationRequestParam outParam = this.mediator.getRegistrationParams();
+            String outIpReal = outParam.getPhoneIp();
+            String broadcastIp = outIpReal.substring(0, outIpReal.lastIndexOf(".")) + "255";
+            InetAddress address = InetAddress.getByName(broadcastIp);
+            WizLightingRequest request = new WizLightingRequest(WizLightingMethodType.registration,
+                    this.mediator.getRegistrationParams());
+            request.setId(0);
+
+            byte[] message = this.converter.transformToByteMessage(request);
+            // logger.trace("Raw packet to send: {}", message);
+
+            // Initialize a datagram packet with data and address
+            DatagramPacket packet = new DatagramPacket(message, message.length, address, DEFAULT_BULB_UDP_PORT);
+
+            // Create a datagram socket, send the packet through it, close it.
+            // For discovery we will "fire and forget" and let the mediator take care of the
+            // responses
+            dsocket = new DatagramSocket();
+            dsocket.send(packet);
+            logger.debug("Sent packet to address: {} and port {}", address, DEFAULT_BULB_UDP_PORT);
+        } catch (IOException exception) {
+            logger.debug("Something wrong happened when broadcasting the packet to port {}... msg: {}",
+                    DEFAULT_BULB_UDP_PORT, exception.getMessage());
+        } finally {
+            if (dsocket != null) {
+                dsocket.close();
+            }
+        }
+    }
+
+    /**
+     * Method called by mediator, after receiving a packet from an unknown WiZ bulb
+     *
+     * @param bulbMacAddress the mac address from the device.
+     * @param bulbIpAddress the host address from the device.
+     */
+    public void discoveredLight(final String lightMacAddress, final String lightIpAddress) {
+        Map<String, Object> properties = new HashMap<>(1);
+        properties.put(CONFIG_MAC_ADDRESS, lightMacAddress);
+        properties.put(CONFIG_IP_ADDRESS, lightIpAddress);
+
+        // NOTE: Only full color bulbs supported at this time
+        ThingUID newThingId = new ThingUID(THING_TYPE_WIZ_COLOR_BULB, lightMacAddress);
+        DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(newThingId).withProperties(properties)
+                .withLabel("WiZ Full Color Bulb at " + lightIpAddress).withRepresentationProperty(lightMacAddress)
+                .build();
+
+        logger.debug("New WiZ bulb sent to inbox! MAC: {}  IP: {}", lightMacAddress, lightIpAddress);
```

Comment:
New inbox discoveries are automatically logged so you shouldn't need to log this.

Reply:
Removed